skills/core.go

```diff
@@ -36,3 +40,37 @@ type Slot struct {
 	Attrs map[string]interface{}
 	Apps  []string
 }
+
+// SecuritySystem is a name of a security system.
+type SecuritySystem string
+
+// Type describes a group of interchangeable capabilities with common features.
+// Types are managed centrally and act as a contract between system builders,
+// application developers and end users.
+type Type interface {
+	// Unique and public name of this type.
+	Name() string
+	// Sanitize checks if a skill is correct, altering if necessary.
+	Sanitize(skill *Skill) error
+	// SecuritySnippet returns the configuration snippet that should be used by
+	// the given security system to enable this skill to be consumed.
+	// An empty snippet is returned when the skill doesn't require anything
+	// from the security system to work, in addition to the default configuration.
+	// ErrUnknownSecurity is returned when the skill cannot deal with the
+	// requested security system.
+	SecuritySnippet(skill *Skill, securitySystem SecuritySystem) ([]byte, error)
+}
+
+const (
```

Comment:
I guess I'm in for the nitpick award of the week here, but I would swap places of "type SecuritySystem" and "type Type" so that SecuritySystem is closer to where it is used.